### PR TITLE
update method of getting avatar's contact

### PR DESCRIPTION
**Pull request by asmaNode**
**Tasks:**

- Update method of displaying conatct-s avatars
- Resolve tests errors

**Demo**
Avatars are displaying correctly

demo_contact_avatar

### CODE REVIEW

File: src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 var expect = chai.expect;
 
-describe('The contactEditionForm directive', function() {
+describe.only('The contactEditionForm directive', function() {
```
Comment (rezk2ll) [inline src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js:7]:
```suggestion
describe('The contactEditionForm directive', function() {
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ require('../shell/shell.service.js');
 require('./contact-shell-builder.service.js');
 require('../vcard/vcard-builder.service.js');
 require('../app.constant.js');
+const _ = require('lodash');
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:5]:
```suggestion
```

not needed

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 3]
```diff
@@ -615,5 +622,39 @@ require('../app.constant.js');
...
+      * @return {Promise}          If success :  resolve avatar data
+      *                            If not reject error
+      */
+
+    function getAvatar(payload) {
+      return $http({ method: 'GET', responseType: 'blob', url: `${getAvatarUrl(payload)}` })
+        .then(response => readFile(response.data, function(e) {
+          return e.target.result;
+        })).catch(error => error);
+    }
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:642]:
```suggestion
/**
 * Get the avatar of the user
 *
 * @param {Ojbect} payload  - The contact information
 *
 * @returns {Promise}       - Resolves with the avatar or rejects with an error
 */
function getAvatar(payload) {
  return $http({ method: 'GET', responseType: 'blob', url: `${getAvatarUrl(payload)}` })
    .then(({ data }) => readFile(data, ({ target }) => target.result))
    .catch(console.error);
}
```

File: src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js (hunk 1 above)
Comment (asmaNode) [reply src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js:7]: tests still in progress

**Review by alagane — APPROVED** [review]

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js (hunk 3 above)
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:642]:
```suggestion
    function getAvatar(payload) {
      return contactRestangularService
        .one('contacts')
        .one(payload.addressBookId)
        .one(payload.addressbookName)
        .one(payload.contactId)
        .one('avatar')
        .get()
        .then(({ data }) => ReadFile(data, e => e.target.result))
        .catch(console.error);
    }
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 4]
```diff
@@ -615,5 +622,39 @@ require('../app.constant.js');
...
+
+    function getAvatarUrl(payload) {
+      return httpConfigurer.getUrl(`${BASE_API_PATH}/contacts/${payload.addressBookId}/${payload.addressbookName}/${payload.contactId}`);
+    }
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:646]:
```suggestion
```

File: src/esn.contact.libs/app/addressBook/addressbook-cache.service.spec.js [hunk 5]
```diff
@@ -6,14 +6,15 @@ var expect = chai.expect;
...
-  var ContactAPIClientMock;
+  let ContactAPIClientMock;
   var BOOK_ID = '123456789';
   var BOOK_NAME = 'contacts';
 
   beforeEach(function() {
     ContactAPIClientMock = {};
     angular.mock.module('linagora.esn.contact', function($provide) {
       $provide.value('ContactAPIClient', ContactAPIClientMock);
+
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/addressbook-cache.service.spec.js:17]:
```suggestion
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 6]
```diff
@@ -615,5 +620,39 @@ require('../app.constant.js');
...
+      * Get contact Avatar
+      * @param  {Object} payload   The contact Id
+      * @param  {Object} options   Includes:
+      *                               + addressBookId
+      *                               + addressbookName
+      *                               + contactId
+      * @return {Promise}          If success :  resolve avatar data
+      *                            If not reject error
+      */
+
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:634]:
```suggestion
  /**
    * Get contact Avatar
    * @param  {Object} payload   The contact Id
    * @param  {Object} options   Includes:
    *                               + addressBookId
    *                               + addressbookName
    *                               + contactId
    * @return {Promise}          If success :  resolve avatar data
    *                            If not reject error
    */
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 7]
```diff
@@ -615,5 +620,39 @@ require('../app.constant.js');
...
+    function getAvatar(payload) {
+      return contactRestangularService
+        .one('contacts/' + payload.addressBookId + '/' + payload.addressbookName + '/' + payload.contactId)
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:637]:
```suggestion

        .one('contacts')
        .one(payload.addressBookId)
        .one(payload.addressbookName)
        .one(payload.contactId)
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.spec.js [hunk 8]
```diff
@@ -4,13 +4,13 @@
 
 var expect = chai.expect;
 
-describe('The contact Angular module contactapis', function() {
+describe.only('The contact Angular module contactapis', function() {
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.spec.js:7]:
```suggestion
describe('The contact Angular module contactapis', function() {
```

File: src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js [hunk 9]
```diff
@@ -1,15 +1,21 @@
 'use strict';
 
 /* global chai: false */
+/* global sinon: false */
```
Comment (rezk2ll) [inline src/linagora.esn.contact/app/contact/form/contact-edition-form.directive.spec.js:4]:
```suggestion
/* global chain, sinon: false */
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js (hunk 7 above)
Comment (asmaNode) [reply src/esn.contact.libs/app/addressBook/contact-api-client.service.js:637]: we can't mock RestAngular with this way , it always generate error when using .one () .one ()...

File: src/linagora.esn.contact/app/contact/list/contact-list-item.directive.spec.js [hunk 10]
```diff
@@ -66,6 +79,15 @@ describe('The contactListItem directive', function() {
...
 
+  it('should display avatar of contact', function() {
+    var phone = '+33333333';
+
+    $scope.contact.tel = [{ type: 'home', value: 'homephone' }, { type: 'work', value: phone }];
+    var element = initDirective();
+
+    expect(element[0].outerHTML).to.contain(phone);
+  });
+
```
Comment (alagane) [inline src/linagora.esn.contact/app/contact/list/contact-list-item.directive.spec.js:90]: This is the same than the previous test, you probably forgot to change it.

**Review by alagane — CHANGES_REQUESTED** [review]

**Review by alagane — DISMISSED** [review]

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 11]
```diff
@@ -29,12 +29,14 @@ require('./contact-dav-client.service.js');
...
     GRACE_DELAY,
     ICAL,
+    contactRestangularService,
     CONTACT_ADDRESSBOOK_DAV_PROPERTIES
   ) {
     var ADDRESSBOOK_PATH = '/addressbooks';
 
     return {
       addressbookHome,
+      getAvatar
```
Comment (alagane) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:39]:
```suggestion
      getAvatar,
```

File: src/esn.contact.libs/app/addressBook/contact-api-client.service.js [hunk 12]
```diff
@@ -656,6 +658,38 @@ require('./contact-dav-client.service.js');
...
+    function readFile(file) {
+      return new Promise((resolve, reject) => {
+        const fr = new FileReader();
+
+        fr.onload = () => {
+          resolve(fr.result);
+        };
+        fr.onerror = reject;
+        fr.readAsDataURL(file);
+      });
```
Comment (alagane) [inline src/esn.contact.libs/app/addressBook/contact-api-client.service.js:691]:
```suggestion
      });
    }
```

File: src/linagora.esn.contact/app/contact/photo/contact-photo-child.pug [hunk 13]
```diff
@@ -1,5 +1,5 @@
 a
-  img(ng-class="{'esn-avatar': listView, 'clickable': editable}" ng-src='{{displayShell.getAvatar(avatarSize)}}' fallback-src="{{displayShell.getDefaultAvatar()}}", alt="{{ 'Profile Picture' | translate }}")
+  img(ng-class="{'esn-avatar': listView, 'clickable': editable}" ng-src="{{contactUserAvatar}}", src="{{displayShell.getDefaultAvatar()}}", alt="{{ 'Profile Picture' | translate }}")
```
Comment (alagane) [inline src/linagora.esn.contact/app/contact/photo/contact-photo-child.pug:2]:
```suggestion
  img(ng-class="{'esn-avatar': listView, 'clickable': editable}" ng-src="{{contactUserAvatar}}" src="" fallback-src="{{displayShell.getDefaultAvatar()}}" alt="{{ 'Profile Picture' | translate }}")
```

**Comment by alagane** [comment]
Replaced by https://github.com/linagora/esn-frontend-contacts/pull/189

**Pull request closed without merging**